File: 3.py
def find_max_weight_node(T, testcases):
    result = []
    
    for _ in range(T):
        N, edges = testcases.pop(0), testcases.pop(0)
        weights = [0] * N
        
        # Calculate weights of each node
        for i in range(N):
            if edges[i] != -1:
                weights[edges[i]] += i
        
        # Find the maximum weight and its corresponding node number
        max_weight = max(weights)
        max_node = weights.index(max_weight)
        
        result.append(max_node)
    
    return result

T=int(input())
testcases=[]
for i in range(T):
    inp = int(input())
    lst1= list(map(int, input().split()))
    testcases.append(inp)
    testcases.append(lst1)

lst=find_max_weight_node(T, testcases)  # Output: [22]
for i in lst:
    print(i , end=" ")

File: 3.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> find_max_weight_node(int T, vector<vector<int>> &testcases)
{
    vector<int> result;

    for (int t = 0; t < T; ++t)
    {
        int N = testcases[2 * t][0];
        vector<int> edges = testcases[2 * t + 1];
        vector<int> weights(N, 0);

        // Calculate weights of each node
        for (int i = 0; i < N; ++i)
        {
            if (edges[i] != -1)
            {
                weights[edges[i]] += i;
            }
        }

        // Find the maximum weight and its corresponding node number
        int max_weight = *max_element(weights.begin(), weights.end());
        int max_node = distance(weights.begin(), find(weights.begin(), weights.end(), max_weight));

        result.push_back(max_node);
    }

    return result;
}

int main()
{
    int T;
    cin >> T;
    vector<vector<int>> testcases(2 * T);

    for (int i = 0; i < T; ++i)
    {
        int inp;
        cin >> inp;
        testcases[2 * i].push_back(inp);

        vector<int> lst1(inp);
        for (int j = 0; j < inp; ++j)
        {
            cin >> lst1[j];
        }
        testcases[2 * i + 1] = lst1;
    }

    vector<int> lst = find_max_weight_node(T, testcases);
    for (int i : lst)
    {
        cout << i << " ";
    }

    return 0;
}
